tests/test_facebook_bookmarks.py

```diff
@@ -43,73 +45,224 @@ def expected_sync_streams():
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        ### Update State Between Syncs
+        ##########################################################################
+
+        new_states = {'bookmarks': dict()}
+        simulated_states = self.calculated_states_by_stream(first_sync_bookmarks)
+        for stream, new_state in simulated_states.items():
+            replication_key = list(expected_replication_keys[stream])[0]
+            new_states['bookmarks'][stream] = {replication_key: new_state}
+        menagerie.set_state(conn_id, new_states)
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+
+        ##########################################################################
+        ### Test By Stream
+        ##########################################################################
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_replication_method = expected_replication_methods[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+
+                if expected_replication_method == self.INCREMENTAL:
+
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = list(expected_replication_keys[stream])[0]
+                    self.assertEqual(1, len(list(expected_replication_keys[stream]))) # catches unexpected compound replication key
```

Comment:
Why do we cast to `list` first? Isn't `expected_replication_keys[stream]` a set?

Reply:
Just wanted to grab the first item. Could do next(iter(expected_replication_keys[stream])) instead.

Reply:
Done.